examples/scripts/ConfigureRemotingForAnsible.ps1

```diff
@@ -93,7 +93,7 @@ Function New-LegacySelfSignedCert
     )
 
     $hostnonFQDN = $env:computerName 
-    $hostFQDN = [System.Net.Dns]::GetHostByName(($env:computerName)).Hostname
+    $hostFQDN = (Get-WmiObject win32_computersystem).DNSHostName+"."+(Get-WmiObject win32_computersystem).Domain.ToLower()
```

Comment:
This has 2 issues

* In the case of the host not being in a domain, this would results in `hostname.workgroup` which isn't really correct?
* We should be using the CIM cmdlets over the WMI ones, e.g. `Get-CimInstance -ClassName Win32_ComputerSystem`

I'm thinking we could do something like the following

```
$hostFQDN = $null
$computer_info = Get-CimInstance -ClassName Win32_ComputerSystem -Property DNSHostName, Domain, DomainRole
# 0 is workstation standalone and 2 is server standalone
if ($computer_info.DomainRole -notin @(0, 2)) {
    $hostFQDN = "$($computer_info.DNSHostName).$($computer_info.Domain)".ToLower()
}
```

From there we then only add it to the alternative names array further down if `hostFQDN` is not null.